### ISSUE TITLE
`test_model_with_brackets_in_shape` test  in vis_utils_test.py is failing

### ISSUE THREAD

**Issue by shilpakancharla** [keras-team/keras#17438]
**Describe the problem**.

Upon running the `test_model_with_brackets_in_shape` test within vis_utils_test.py in Google Colab (shown below), it was found that the test would fail. Moreover, whenever the `show_shapes=True` parameter is passed in, this test fails. If this value is False, then it is able to plot the model.

```
import os
import tensorflow as tf
import tensorflow_hub as hub
keras = tf.keras

class DictLayer(keras.layers.Layer):
    def call(self, inputs) -> tf.Tensor:
        tensor_input, dict_input = inputs
        return tf.concat(list(dict_input.values()), axis=1)

inputs = {
    "a": keras.Input(name="a", shape=(1), dtype=tf.float32),
    "b": keras.Input(name="b", shape=(1), dtype=tf.float32),
}
outputs = DictLayer()((inputs["a"], inputs))
model = keras.Model(
    inputs=inputs,
    outputs=outputs,
)
keras.utils.plot_model(
    model, show_shapes=True, show_dtype=True, show_layer_names=True
)
```

Essentially, any model that has an output that takes in a dictionary input fails. 

**Describe the current behavior**.

Upon running the code snippet above (shown in https://colab.corp.google.com/drive/1zYeZeZcSTCbXRQPqwABumvMFEoeMaZ2G?usp=sharing), the following error is received: 

```
InvocationException: Program terminated with status: 1. stderr follows: Error: bad label format {dict_layer_3|DictLayer|float32}|{input:|output:}|{{((None, 1), {'a': (None, 1), 'b': (None, 1)})}|{(None, 2)}}
```

**Describe the expected behavior**.

We expect the type of model above, specifically which takes a dictionary as the output, should be plotted by `plot_model` when `show_shapes=True`. 

**Standalone code to reproduce the issue**.

Colab: https://colab.corp.google.com/drive/1zYeZeZcSTCbXRQPqwABumvMFEoeMaZ2G?usp=sharing

**Comment by old-school-kid** [comment]
M nnnn

**Comment by ianstenbit** [comment]
Strangely, this error is _not_ being thrown in our CI environment, but I am able to reproduce it on my personal Linux environment.

I think that the issue may be related to `graphviz` version, but I'm not certain.

**Comment by ianstenbit** [comment]
It looks like Keras may actually be using the wrong escape character for curly brackets.
(See https://graphviz.org/doc/info/shapes.html#record -- it looks like the backslash is the correct escape character, not the forward slash, which is what vis_utils is currently using).

I will raise a PR to fix this.

**Comment by ianstenbit** [comment]
This should be fixed by #17483. Feel free to re-open and assign to me if you continue to see issues.
